src/entity/object/ItemEntity.php

```diff
@@ -226,12 +226,18 @@ public function onCollideWithPlayer(Player $player) : void{
 		}
 
 		$item = $this->getItem();
-		$playerInventory = $player->getInventory();
+		$playerInventory = match(true){
+			$player->getOffHandInventory()->canAddItem($item) and !$player->getOffHandInventory()->isSlotEmpty(0) => $player->getOffHandInventory(),
```

Comment:
It took me a couple of minutes to understand what this branch is doing.
A more self-explanatory check might be `$player->getOffHandInventory()->getItem(0)->canStackWith($item)`.